sw/airborne/arch/sim/mcu_periph/i2c_arch.c

```diff
@@ -27,16 +27,22 @@
 #include "mcu_periph/i2c.h"
 
 
-bool i2c_idle(struct i2c_periph *p __attribute__((unused))) { return true; }
-bool i2c_submit(struct i2c_periph *p __attribute__((unused)), struct i2c_transaction *t __attribute__((unused))) { return true;}
+static bool i2c_sim_idle(struct i2c_periph *p __attribute__((unused))) __attribute__((unused));
+static bool i2c_sim_submit(struct i2c_periph *p __attribute__((unused)), struct i2c_transaction *t __attribute__((unused))) __attribute__((unused));
+static void i2c_sim_setbitrate(struct i2c_periph *p __attribute__((unused)), int bitrate __attribute__((unused))) __attribute__((unused));
```

Comment:
gcc did not accept the unused attribute in the function definition, only in the declaration. Turns out it needs to be placed in front of the function name instead of behind it. Fixed.